src/features/antora/antoraSupport.ts

```diff
@@ -83,18 +86,141 @@ export async function getAntoraConfig (textDocument: vscode.TextDocument): Promi
+}
+
+class AntoraDisabledError extends Error {
+}
+
+export async function parseAntoraConfig (textDocumentUri: Uri): Promise<{ [key: string]: any }> {
+  const antoraConfigUri = await getAntoraConfig(textDocumentUri)
+  if (antoraConfigUri !== undefined) {
+    const antoraConfigPath = antoraConfigUri.fsPath
+    try {
+      return yaml.load(fs.readFileSync(antoraConfigPath, 'utf8'))
+    } catch (err) {
+      console.log(`Unable to parse ${antoraConfigPath}, cause:` + err.toString())
+      return {}
+    }
+  }
+  return {}
+}
+
+export async function getAttributes (textDocumentUri: Uri): Promise<{ [key: string]: string }> {
+  const doc = await parseAntoraConfig(textDocumentUri)
+  if (doc !== {}) {
     return doc.asciidoc.attributes
-  } catch (err) {
-    console.log(`Unable to parse ${antoraConfigPath}, cause:` + err.toString())
+  } else {
     return {}
   }
 }
 
-export async function getValidConfig (textDocument: vscode.TextDocument): Promise<boolean> {
-  return await getAntoraConfig(textDocument) !== undefined
+export async function getContentCatalog (textDocumentUri: Uri, extensionContext: ExtensionContext): Promise<ContentCatalog | undefined> {
+  try {
+    const playbook = await createPlaybook(textDocumentUri, extensionContext)
+    if (playbook !== undefined) {
+      const contentAggregate = await aggregateContent(playbook)
+      return classifyContent(playbook, contentAggregate)
+    }
+    return undefined
+  } catch (e) {
+    if (e instanceof AntoraDisabledError) {
+      return undefined
+    } else {
+      console.log(`Unable to create contentCatalog : ${e}`)
+      throw e
+    }
+  }
+}
+
+async function createPlaybook (textDocumentUri: Uri, extensionContext: ExtensionContext): Promise<{
+  site: {};
+  runtime: {};
+  content: {
+    sources: {
+      startPath: string;
+      branches: string;
+      url: string
+    }[]
+  }
+} | undefined> {
+  const activeAntoraConfig = await getActiveAntoraConfig(textDocumentUri, extensionContext)
+  if (activeAntoraConfig === undefined) {
+    return undefined
+  }
+  const contentSourceRootPath = path.dirname(activeAntoraConfig.fsPath)
+  const contentSourceRepositoryRootPath = workspace.getWorkspaceFolder(activeAntoraConfig).uri.fsPath
+  // https://docs.antora.org/antora/latest/playbook/content-source-start-path/#start-path-key
+  const startPath = path.relative(contentSourceRepositoryRootPath, contentSourceRootPath)
+  return {
+    content: {
+      sources: [{
+        url: contentSourceRepositoryRootPath,
+        branches: 'HEAD',
+        startPath,
+      }],
+    },
+    runtime: {},
+    site: {},
+  }
+}
+
+export async function getSrc (textDocumentUri: Uri, contentCatalog: ContentCatalog | undefined):Promise<{ [key: string]: any } | {}> {
+  const antoraConfig = await getAntoraConfig(textDocumentUri)
+  const doc = await parseAntoraConfig(textDocumentUri)
+  if (antoraConfig !== undefined) {
+    const contentSourceRootPath = path.dirname(antoraConfig.fsPath)
```

Comment:
we don't need to compute `contentSourceRootPath` if `doc` is equals to `{}` or if the `contentCatalog` is undefined.
We also don't need to resolve/parse the Antora configuration since we don't have a content catalog.